tests/conditional_compilation/test_infer.py

```diff
@@ -4,19 +4,36 @@
 
 """ Test inference with conditional compiled binaries.
 """
+
 import sys
+import os
+from inspect import getsourcefile
+from pathlib import Path
 from proc_utils import cmd_exec  # pylint: disable=import-error
 from install_pkg import get_openvino_environment  # pylint: disable=import-error
 
 
-def test_infer(test_id, model, artifacts):
-    """ Test inference with conditional compiled binaries
-    """
+def run_infer(artifacts, test_id, model, out):
     install_prefix = artifacts / test_id / "install_pkg"
-    exe_suffix = ".exe" if sys.platform == "win32" else ""
-    benchmark_app = install_prefix / "bin" / f"benchmark_app{exe_suffix}"
-    returncode, _ = cmd_exec(
-        [str(benchmark_app), "-d=CPU", f"-m={model}", "-niter=1", "-nireq=1"],
+    returncode, output = cmd_exec(
+        [sys.executable, str((Path(getsourcefile(lambda: 0)) / ".." / "tools" / "infer_tool.py").resolve()),
+         "-d=CPU", f"-m={model}", f"-r={out}"],
         env=get_openvino_environment(install_prefix),
     )
-    assert returncode == 0, f"Command exited with non-zero status {returncode}"
+    return returncode, output
+
+
+def test_infer(test_id, model, artifacts, openvino_root_dir, openvino_cc):
+    """ Test inference with conditional compiled binaries
+    """
+    tmp = os.environ["PATH"]
```

Comment:
Test shouldn't change env variables, but `cmd_exec` should be called with a specific environment